### DISCUSSION TITLE
Grapher local setup issue: buildTsc and testJest failing

### DISCUSSION THREAD

**Discussion by rhys-the-davies** [owid/owid-grapher#4335] [Q&A]
Hello there o/ sorry if this is a stupid newbie issue, I hope it is. 

When I follow the Quick start ['local development setup'](https://github.com/owid/owid-grapher/blob/master/docs/docker-compose-mysql.md) guide I get to [installing node.js and yarn](https://github.com/owid/owid-grapher/blob/master/docs/local-typescript-setup.md) and everything works grea tuntil I run `yarn buildTsc` and `yarn testJest`.

buildTsc finds 1062 errors of the like in the screenshot: 
<img width="968" alt="Screenshot 2024-12-18 at 21 52 30" src="https://github.com/user-attachments/assets/11d2461c-7059-48fe-a4f0-0df1bc4b1445" />

And testJest fails presumably because of those errors.

I'm on a Mac with Apple Silicon and I tried in and out of a docker container. 

I tried searching around and figuring it out but I'm not a developer and think I'm missing something obvious. I'd love to be able to get this working to use the grapher in a series of articles I'm writing inspired by ourworldindata.

**Comment by marcelgerber** [comment]
Hi Rhys!
Sorry, these docs were indeed outdated. I've just updated them - the command you need to run here is `yarn lerna run build && yarn buildTsc`.

In general, our recommended setup is to [run inside Docker](https://github.com/owid/owid-grapher/blob/master/docs/docker-compose-mysql.md), where `make up` should hopefully take care of most steps.

Also, could you elaborate a bit on your intended usage of Grapher? We're always curious to see some of the things other people are using it for.

**Reply by rhys-the-davies** [reply]
Hey Marcel o/

Thanks for your help here, much appreciated 😊 That now works 👍🏼 

I will say in case you have ideas, I then run into the issue that perhaps isn't one you can fix 

- Homebrew's installation script doesn't seem to work on ARM silicon. I tried cloning the repo and set up the Homebrew environment but `brew install tmux` tells me ruby is missing
- Ruby also isn't playing with apple silicon so I modified the devcontainer to have the ruby prerequisites, cloned the rbenv repo and installed the ruby-build plugin and installed ruby that way
- But installing tmux then tells me `Error: tmux: no bottle available!` with a bunch of text from the developers about building from source = bad 

I don't suppose you or anyone else has it working on an M3 chip and I'm just missing something? 

Also, yes, I'd like to use grapher for two things:
1. I'm writing a blog series called 'inspirational graphs' to talk about the graphs that interesting people I know find inspirational and I'd like to re-create each of them using grapher
2. I like to write and run fact-checks on some of the crazy stuff people say wiht the caveat of 'people say' or 'i heard' and I think I'd like to use it when I need it there too.
